src/ImageSharp/Image.WrapMemory.cs

```diff
@@ -18,6 +18,14 @@ public abstract partial class Image
         /// <summary>
         /// Wraps an existing contiguous memory area of 'width' x 'height' pixels,
         /// allowing to view/manipulate it as an <see cref="Image{TPixel}"/> instance.
+        /// The ownership of the underlying buffer for the input <see cref="Memory{T}"/> instance is not being transferred to
+        /// the new <see cref="Image{TPixel}"/> instance, meaning that consumers of this method need to make sure the input buffer
+        /// is either self-contianed (for instance, this is the case for a <see cref="Memory{T}"/> instance wrapping a new array that was
+        /// created), or that the owning object is not disposed until the returned <see cref="Image{TPixel}"/> is not disposed.
+        /// For instance, if the input <see cref="Memory{T}"/> instance is one retrieved from an <see cref="IMemoryOwner{T}"/> instance
+        /// rented from a memory pool (such as <see cref="MemoryPool{T}"/>), and that owning instance is disposed while the image is still
+        /// in use, this will lead to undefined behavior and possibly runtime crashes (as the same buffer might then be modified by other
+        /// consumers while the returned image is still working on it). Make sure to control the lifetime of the input buffers properly.
         /// </summary>
         /// <typeparam name="TPixel">The pixel type</typeparam>
```

Comment:
```xml
/// <summary>
/// <para>
/// Wraps an existing contiguous memory area of 'width' x 'height' pixels allowing viewing/manipulation as
/// an <see cref="Image{TPixel}"/> instance.
/// </para>
/// <para>
/// Please Note: Using this method does not transfer the ownership of the underlying buffer of the input <see cref="Memory{T}"/>
/// to the new <see cref="Image{TPixel}"/> instance. This means that consumers of this method must ensure that the input buffer
/// is either self-contained, (for example, a <see cref="Memory{T}"/> instance wrapping a new array that was
/// created), or that the owning object is not disposed until the returned <see cref="Image{TPixel}"/> is disposed.
/// </para>
/// <para>
/// If the input <see cref="Memory{T}"/> instance is one retrieved from an <see cref="IMemoryOwner{T}"/> instance
/// rented from a memory pool (such as <see cref="MemoryPool{T}"/>), and that owning instance is disposed while the image is still
/// in use, this will lead to undefined behavior and possibly runtime crashes (as the same buffer might then be modified by other
/// consumers while the returned image is still working on it). Make sure to control the lifetime of the input buffers appropriately.
/// </para>
/// </summary>
```